hr_worked_days_from_timesheet/models/hr_payslip.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding:utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2012 - 2014 Odoo Canada. All Rights Reserved.
+#    Copyright (C) 2015 Acysos S.L. All Rights Reserved.
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as published
+#    by
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from openerp import models, api
+from openerp.tools.translate import _
+from datetime import datetime
+from openerp.tools import DEFAULT_SERVER_DATE_FORMAT
+
+
+class HrPayslip(models.Model):
+    _inherit = 'hr.payslip'
+
+    def timesheet_mapping(self, timesheet_sheets, payslip, date_from, date_to,
+                          date_format):
+        """This function takes timesheet objects imported from the timesheet
+        module and creates a dict of worked days to be created in the payslip.
+        """
+        # Create one worked days record for each timesheet sheet
+        wd_model = self.env['hr.payslip.worked_days']
+        for ts_sheet in timesheet_sheets:
+            # Get formated date from the timesheet sheet
+            date_from_formated = datetime.strptime(
```

Comment:
You could use fields.Datetime.from_string and fields.Date.to_string functions available in new api.